### PR TITLE
Checkout: Always use foldable card for non-critical Terms of Service

### LINKED ISSUE #87368: Checkout: `Read more` is still visible when the container is empty
Image

As I was testing TOS updates, I noticed that the `Read more` button is still visible even if the container is empty. I would expect that if there were no additional TOS statements in the container, the button would be hidden.

I was forcing the promotional period TOS by hard-coding a discount in the backend. So this probably needs to be tested with a 100% coupon and other products to confirm if the behavior is consistent.

### PR DESCRIPTION
**Pull request by sirbrillig**
Checkout has a very large number of "terms of service" (TOS) notices which are displayed at the bottom of the checkout form before the payment button. These notices can in some cases take a up large amount of vertical space and make the checkout experience feel cluttered. To help improve this situation, an AB test was run to determine how useful it would be to hide the non-essential TOS notices behind a foldable card that says "Read more". This was only for TOS notices which were not legally required to be visible without a click. The test showed no danger with this approach and so it was launched to most checkout sections in https://github.com/Automattic/wp-calypso/pull/85445. Additional sections were launched in https://github.com/Automattic/wp-calypso/pull/86215.

As we near the potential launch of the full checkout v2 design, I believe that we should launch the foldable card for all instances of checkout.

## Proposed Changes

This PR activates the foldable "Read more" card for all versions of checkout. Specifically, this launches the new UX for Jetpack checkout and Akismet checkout which were the only two remaining holdouts.

Note that in most cases, Jetpack and Akismet checkout currently do not actually have anything in the "Read more" card and so they will not show it. The main change therefore will be the small UX differences like adding the refund icon and removing the "i" icon. See the screenshots below.

As a side effect, this fixes https://github.com/Automattic/wp-calypso/issues/87368.

## Screenshots

What | Before             |  After
:-----:|:-------------------------:|:-------------------------:
Akismet | before-87554-akismet | after-87554-akismet
Jetpack plan | before-87554-jetpack-plan | after-87554-jetpack-plan
Jetpack one-time | before-87554-jetpack-one-time | after-87554-jetpack-one-time

## Testing Instructions

- Visit checkout for a Jetpack site (eg: `/checkout/jetpack/jetpack_complete`) or for an Akismet product (eg: `/checkout/akismet/ak_plus_yearly_2`).
- Verify that the UX of the TOS at the bottom of the page looks reasonable.

### CODE REVIEW

File: client/my-sites/checkout/src/components/payment-method-step.tsx [hunk 1]
```diff
@@ -15,16 +15,12 @@ import styled from '@emotion/styled';
...
 
-const CheckoutTermsWrapper = styled.div< {
-	showToSFoldableCard: boolean;
-} >`
+const CheckoutTermsWrapper = styled.div`
 	& > * {
 		margin: 16px 0;
-		padding-left: ${ ( { showToSFoldableCard } ) =>
-			hasCheckoutVersion( '2' ) || showToSFoldableCard ? null : '24px' };
+		padding-left: none;
```
Comment (jjchrisdiehl) [inline client/my-sites/checkout/src/components/payment-method-step.tsx]: This should be `padding-left: 0`

File: client/my-sites/checkout/src/components/checkout-terms.tsx [hunk 2]
```diff
@@ -91,54 +82,53 @@ export default function CheckoutTerms( { cart }: { cart: ResponseCart } ) {
...
+function CheckoutTermsReadMore( { children }: { children: ReactNode } ) {
+	const translate = useTranslate();
+	// Note that this technique for finding children does not work for strings or
+	// empty fragments. Hopefully all children passed to this component are
+	// components themselves or null.
+	if ( ! children || Children.toArray( children ).filter( isValidElement ).length === 0 ) {
+		return null;
+	}
+	return (
+		<TermsCollapsedContent>
```
Comment (jjchrisdiehl) [inline client/my-sites/checkout/src/components/checkout-terms.tsx:120]: Is this being rendered twice? See line 95. Not sure it really matters

**Comment by jjchrisdiehl** [comment]
Overall looks great, just two minor notes.

Functional testing:

**Jetpack**
image


**Akismet**
image


**WPCOM**
image

File: client/my-sites/checkout/src/components/checkout-terms.tsx (hunk 2 above)
Comment (sirbrillig) [reply client/my-sites/checkout/src/components/checkout-terms.tsx:120]: Huh, you're right.

**Pull request merged**